server/main.lua

```diff
@@ -232,44 +252,56 @@ end)
-    xPlayer.Functions.SetCreditCard(cardNumber)
-    local info = {}
-    local selectedCard = Config.cardTypes[math.random(1,#Config.cardTypes)]
-    info.citizenid = cid
-    info.name = xPlayer.PlayerData.charinfo.firstname .. ' ' .. xPlayer.PlayerData.charinfo.lastname
-    info.cardNumber = cardNumber
-    info.cardPin = tonumber(pin)
-    info.cardActive = true
-    info.cardType = selectedCard
-
-    if selectedCard == "visa" then
-        xPlayer.Functions.AddItem('visa', 1, nil, info)
-    elseif selectedCard == "mastercard" then
-        xPlayer.Functions.AddItem('mastercard', 1, nil, info)
+    local auth = validatePosition(GetEntityCoords(GetPlayerPed(src)))
+    if auth.isNearBank then
+        local cid = xPlayer.PlayerData.citizenid
+        local cardNumber = math.random(1000000000000000,9999999999999999)
+        xPlayer.Functions.SetCreditCard(cardNumber)
+        local info = {}
+        local selectedCard = Config.cardTypes[math.random(1,#Config.cardTypes)]
+        info.citizenid = cid
+        info.name = xPlayer.PlayerData.charinfo.firstname .. ' ' .. xPlayer.PlayerData.charinfo.lastname
+        info.cardNumber = cardNumber
+        info.cardPin = tonumber(pin)
+        info.cardActive = true
+        info.cardType = selectedCard
+    
+        if selectedCard == "visa" then
+            xPlayer.Functions.AddItem('visa', 1, nil, info)
+        elseif selectedCard == "mastercard" then
+            xPlayer.Functions.AddItem('mastercard', 1, nil, info)
+        end
+    
+        TriggerClientEvent('qb-banking:openBankScreen', src)
+        TriggerClientEvent('QBCore:Notify', src, Lang:t('success.debit_card'), 'success')
+    
+        TriggerEvent('qb-log:server:CreateLog', 'banking', 'Banking', 'lightgreen', "**"..GetPlayerName(xPlayer.PlayerData.source) .. " (citizenid: "..xPlayer.PlayerData.citizenid.." | id: "..xPlayer.PlayerData.source..")** successfully ordered a debit card")    
+    else
+        TriggerClientEvent('QBCore:Notify', src, Lang:t('error.too_far_away'), "error")
+        TriggerEvent('qb-log:server:CreateLog', 'banking', 'Banking', "red", "**"..GetPlayerName(xPlayer.PlayerData.source) .. " (citizenid: "..xPlayer.PlayerData.citizenid.." | id: "..xPlayer.PlayerData.source..")** has attempted to order a debit card whilst not at a banking location, Nearest bank = "..auth.nearestDistance)
     end
-
-    TriggerClientEvent('qb-banking:openBankScreen', src)
-    TriggerClientEvent('QBCore:Notify', src, Lang:t('success.debit_card'), 'success')
-
-    TriggerEvent('qb-log:server:CreateLog', 'banking', 'Banking', 'lightgreen', "**"..GetPlayerName(xPlayer.PlayerData.source) .. " (citizenid: "..xPlayer.PlayerData.citizenid.." | id: "..xPlayer.PlayerData.source..")** successfully ordered a debit card")
 end)
 
 RegisterNetEvent('qb-banking:doQuickDeposit', function(amount)
     local src = source
     local xPlayer = QBCore.Functions.GetPlayer(src)
     while xPlayer == nil do Wait(0) end
```

Comment:
Remove this while loop and make the if statement check which I added in the suggestion above